### ISSUE TITLE
Any plans for DNX and asp.net core?

### ISSUE THREAD

**Issue by kevin5464** [prismicio/csharp-kit#12]

**Comment by erwan** [comment]
Hi Kevin,

What's do you mean, what's required to be compatible with DNX?

**Comment by kevin5464** [comment]
Well - while I was waiting, I forked the code and pulled it down to recompile as asp.net core (update1). 

In short, a few things changed:
- HttpUtility is now WebUtility
- I removed all Cache options because caching is different in core, and I've already got that working outside of this gateway request pipeline

Essentially, fork the code, load the C# classes into a new asp.net core class library project, and get it to compile and successfully run tests.

**Comment by erwan** [comment]
Is it possible to do so without breaking compatibility for people not using DNX yet?

**Comment by kevin5464** [comment]
Yes - it will compile with either 4.5.1 or 5.0. 

When I get a little free time, I'll do a formal pull request if you're interested in checking it out. 

We're using it right now in our asp.net core app and it's working like a charm - thanks for working on it!

**Comment by erwan** [comment]
It's good to hear!

Thank you for the help, a PR is welcome but we may wait until the official release of Core 1.0 before releasing a version of the kit integrating it.

**Comment by BjornPoppe** [comment]
.NET Core is now released, any updates on a release of csharp-kit for core?

**Comment by erwan** [comment]
Yes, we're planning to do it but we don't have an ETA yet. We're busy on other fronts for now, but eventually we'll tackle this.

**Comment by PhilipBathe** [comment]
Did Kevin5464 ever do that pull request by any chance?

**Comment by snehahebbar** [comment]
Will there be a package compatible with .Net Core?

**Comment by benembery** [comment]
I've forked this and have it working in a production environment with no issues. Though my implementation doesn't you use all the possible fragments. Unfortunately, I have no access to the "Les Bonne Chose" repository so can't run any of the tests. 

I would be uncomfortable submitting a pull request without proof that those tests work.
https://github.com/benembery/csharp-kit

To use this I built the updated package in VS2017 and put the resulting nuGet package and in a `packages` folder in my project. You then need to add a nuget.config that looks something like this. I've tested this building on both a Mac and in Visual Studio Team Services and deployment to Azure.

````xml
<?xml version="1.0" encoding="utf-8"?>
    <configuration>
  <packageSources>
    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetcidev/api/v3/index.json" />
    <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
    <add key="Local" value="./packages" />
  </packageSources>
</configuration>
````

Hope this helps.

**Comment by srenault** [comment]
For you information, we plan to update the kit to use the asp.net core.

**Comment by DavidDeSloovere** [comment]
.NET Standard is what you want to target, that way the lib will run in both .NET core and the full .NET framework. You can go for 1.6 or 2.0. I wouldn't bother with 1.6 anymore though.
More information https://docs.microsoft.com/en-us/dotnet/standard/net-standard

**Comment by benembery** [comment]
@DavidDeSloovere see https://github.com/prismicio/csharp-kit/pull/15. 
I've create a PR to support 1.6 and retain .NET 4.5 backwards compatability. I've build a version of package and I'm using it on a few different projects with no issues.

**Comment by benembery** [comment]
Unfortunately, I lack the knowledge of Travis and time to update the CI build.

**Comment by DavidDeSloovere** [comment]
If prismic wants to up its marketshare, I'm sure they'll fix that.

**Comment by bjomi** [comment]
Please have a look at and merge the pull request mentioned above. Working with packages that doesn't support netstandard is awful. For most developers these days I would say. So please fix this now!

**Comment by benembery** [comment]
This can now be closed.  https://github.com/adaptivewebworks/prismicio-netstandard-kit covers this.
